.github/workflows/node-release.yml

```diff
@@ -0,0 +1,185 @@
+name: node-release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: Choose a release type
+        required: true
+        type: choice
+        options:
+          - prerelease 
+          - patch
+          - minor
+          - major
+
+jobs:
+  bump_version:
+    runs-on: ubuntu-20.04
+
+    defaults:
+      run:
+        working-directory: ./
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          fetch-depth: 0
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Bump Version
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          npm version ${{ github.event.inputs.version }} --preid pre -m "Update node version to (${{ github.event.inputs.version }})"
+          git push
+
+  build:
+    strategy:
+      matrix:
+        os:
+          - ubuntu-20.04
+          - macos-12
```

Comment:
For macos, this can be run on our faster M1 runner as well, if set to [self-hosted, macOS, ARM64]